src/sparse_merkle_tree/smt.rs

```diff
@@ -1583,6 +1653,85 @@ impl SparseMerkleTree {
             Err(_) => Ok(false),
         }
     }
+
+    // remove_keys_from_proof removes keys from proof and returns a new proof without them.
+    pub fn remove_keys_from_proof(
+        proof: &Proof,
+        removed_keys: &[&[u8]],
+    ) -> Result<Proof, SMTError> {
+        let filter_map = Self::prepare_queries_with_proof_map(proof)?;
```

Comment:
Similar to the recent discussion we had in the other PR, does this really need to be a hashmap? or can we just use a vec here?

Reply:
If using array and if we need to loop over it to find duplicates and converting it to some other array later on, do we really gain something in comparison to just use hashmap?

Reply:
I think, in this case it would be better to use `HashMap`, because in the last PR we had some iterations over the queries and we were able to find duplicate keys, but here we need queries at the beginning.

Reply:
ok, sounds good